spiketoolkit/postprocessing/features.py

```diff
@@ -96,149 +115,202 @@ def compute_unit_template_features(waveforms,
-    peak_idx = np.argmax(waveform)
-
-    try:
-        if waveform[peak_idx] > np.abs(waveform[trough_idx]):
-            threshold = waveform[peak_idx] * 0.5
-            thresh_crossing_1 = np.min(
-                np.where(waveform[:peak_idx] > threshold)[0])
-            thresh_crossing_2 = np.min(
-                np.where(waveform[peak_idx:] < threshold)[0]) + peak_idx
-        else:
-            threshold = waveform[trough_idx] * 0.5
-            thresh_crossing_1 = np.min(
-                np.where(waveform[:trough_idx] < threshold)[0])
-            thresh_crossing_2 = np.min(
-                np.where(waveform[trough_idx:] > threshold)[0]) + trough_idx
-
-        halfwidth = (timestamps[thresh_crossing_2] - timestamps[thresh_crossing_1])
-
-    except ValueError:
-
-        halfwidth = np.nan
-
-    return halfwidth * 1e3
-
-
-def compute_unit_template_pt_ratios(waveform):
-    """
-    Peak-to-trough ratio of 1D waveform
-    Inputs:
-    ------
-    waveform : numpy.ndarray (N samples)
-    Outputs:
-    --------
-    PT_ratio : waveform peak-to-trough ratio
-    """
-
-    trough_idx = np.argmin(waveform)
-
-    peak_idx = np.argmax(waveform)
-
-    PT_ratio = np.abs(waveform[peak_idx] / waveform[trough_idx])
-
-    return PT_ratio
-
-
-def compute_unit_template_repolarization_slopes(waveform, timestamps, window=20):
-    """
-    Spike repolarization slope (after maximum deflection point)
-    Inputs:
-    ------
-    waveform : numpy.ndarray (N samples)
-    timestamps : numpy.ndarray (N samples)
-    window : int
-        Window (in samples) for linear regression
-    Outputs:
-    --------
-    repolarization_slope : slope of return to baseline (V / s)
-    """
-
-    max_point = np.argmax(np.abs(waveform))
-
-    waveform = - waveform * (np.sign(waveform[max_point]))  # invert if we're using the peak
-
-    repolarization_slope = linregress(timestamps[max_point:max_point + window], waveform[max_point:max_point + window])[
-        0]
+# def compute_unit_template_duration(waveform, timestamps):
+#     """
+#     Duration (in seconds) between peak and trough
+#     Inputs:
+#     ------
+#     waveform : numpy.ndarray (N samples)
+#     timestamps : numpy.ndarray (N samples)
+#     Outputs:
+#     --------
+#     duration : waveform duration in milliseconds
+#     """
+#
+#     trough_idx = np.argmin(waveform)
+#     peak_idx = np.argmax(waveform)
+#
+#     # to avoid detecting peak before trough
+#     if waveform[peak_idx] > np.abs(waveform[trough_idx]):
+#         duration = timestamps[peak_idx:][np.where(waveform[peak_idx:] == np.min(waveform[peak_idx:]))[0][0]] - \
+#                    timestamps[peak_idx]
+#     else:
+#         duration = timestamps[trough_idx:][np.where(waveform[trough_idx:] == np.max(waveform[trough_idx:]))[0][0]] - \
+#                    timestamps[trough_idx]
+#
+#     return duration * 1e3
 
-    return repolarization_slope * 1e-6
 
+def compute_trough_to_peak_duration(sorting,
```

Comment:
is "peak_to_valey" a more common naming, no ?

Reply:
to be honest, I don't know =). I'm fine with any, I'll follow your suggestions

Reply:
I think both are used...we can discuss about naming later :)